vms/rpcchainvm/factory.go

```diff
@@ -62,6 +62,7 @@ func (f *factory) New(log logging.Logger) (interface{}, error) {
 	}
 
 	vm := NewClient(vmpb.NewVMClient(clientConn))
+	vm.conns = append(vm.conns, clientConn)
```

Comment:
Nice catch.

It does feel a bit weird that we modify `conns` here though. Would it make more sense if we passed the `clientConn` into `NewClient` (or had a different creation function or something)? It just feels like this is part of the creation of the struct to me

Reply:
yup will update